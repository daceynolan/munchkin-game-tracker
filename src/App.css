.game-header,
.game-content {
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 100%;
}

.game-content {
  margin-top: 10px;
}
