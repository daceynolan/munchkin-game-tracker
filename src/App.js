import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

import SplashScreen from "./components/SplashScreen";
import ConfigureGame from "./components/ConfigureGame";
import Game from "./components/Game";

const App = () => {
  return (
    <Router>
      <div className="App">
        <Switch>
          <Route exact path="/" component={SplashScreen} />
          <Route path="/configure" component={ConfigureGame} />
          <Route path="/game" component={Game} />
        </Switch>
      </div>
    </Router>
  );
};

export default App;
