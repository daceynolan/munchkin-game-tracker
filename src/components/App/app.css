.game-content {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-top: 10px;
  padding: 5px;
}
