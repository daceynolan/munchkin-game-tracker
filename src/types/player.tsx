import { AvatarId } from "types/avatars";

export type Sex = "M" | "F";

export type Player = {
  id: string;
  name: string;
  sex: Sex;
  originalSex: Sex;
  level: number;
  bonus: number;
  avatar: AvatarId;
};
