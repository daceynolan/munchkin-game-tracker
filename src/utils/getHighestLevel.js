export default (levels) => {
  return Math.max(...levels);
};
