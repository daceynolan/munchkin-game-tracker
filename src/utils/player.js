import nanoid from "nanoid";
import defaultAvatar from "../images/dragon.png";

export function Player(name, level, bonus, avatar) {
  this.id = nanoid();
  this.name = name || "";
  this.level = level || 1;
  this.bonus = bonus || 0;
  this.avatar = avatar || {
    alt: "dragon",
    src: defaultAvatar
  };
}
