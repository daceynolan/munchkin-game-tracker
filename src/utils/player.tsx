import nanoid from "nanoid";

import { Player as PlayerType } from "types/player";

export const Player = (function Player(this: PlayerType, player?: PlayerType) {
  const { name = "", sex = "M", level = 1, bonus = 0, avatar = "dragon" } =
    player || {};
  this.id = nanoid();
  this.name = name;
  this.sex = sex;
  this.originalSex = sex;
  this.level = level;
  this.bonus = bonus;
  this.avatar = avatar;
} as any) as { new (player?: PlayerType): PlayerType };
