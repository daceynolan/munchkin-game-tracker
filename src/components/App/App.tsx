import React, { useState, useEffect } from "react";
import { HashRouter as Router, Switch, Route } from "react-router-dom";

import "./app.css";
import Card from "../Card";
import Combat from "../Combat";
import ConfigureGame from "../ConfigureGame";
import CreatePlayer from "../CreatePlayer";
import EditPlayer from "../EditPlayer";
import Game from "../Game";
import GithubLink from "../GithubLink/GithubLink";
import { GameContext } from "../../contexts/gameContext";
import Header from "../Header";
import NotFoundPage from "../NotFoundPage";
import SplashScreen from "../SplashScreen";
import { Player } from "../../utils/player";

const App = () => {
  const [players, setPlayers] = useState(
    //version local storage key to prevent stale data with app changes
    JSON.parse(localStorage.getItem("game") || "") || //change to game_v1.0.0
      Array(2)
        .fill(null)
        .map(() => new Player())
  );

  useEffect(() => {
    localStorage.setItem("game", JSON.stringify(players));
  }, [JSON.stringify(players)]); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <div className="game-content">
      <Header />
      <GithubLink />
      <GameContext.Provider
        value={{
          usePlayers: [players, setPlayers],
        }}
      >
        <Router>
          <Card>
            <Switch>
              <Route exact path="/" component={SplashScreen} />
              <Route path="/configure" component={ConfigureGame} />
              <Route path="/game" component={Game} />
              <Route path="/combat" component={Combat} />
              <Route path="/player/edit/:id" component={EditPlayer} />
              <Route path="/player/create" component={CreatePlayer} />
              <Route path="*" component={NotFoundPage} />
            </Switch>
          </Card>
        </Router>
      </GameContext.Provider>
    </div>
  );
};

export default App;
