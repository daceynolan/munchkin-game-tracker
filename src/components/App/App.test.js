import React from "react";
import { render } from "@testing-library/react";

import App from ".";
import { Player } from "../../utils/player";

const PLAYERS = [new Player({ name: "Dacey" }), new Player({ name: "Tom" })];

describe("<App/>", () => {
  test("renders without crashing", () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("saves game to local storage on change", () => {
    JSON.parse = jest.fn();
    const localStorageGetMock = jest.fn().mockResolvedValue(PLAYERS);
    const localStorageSetMock = jest.fn();
    Object.defineProperty(window, "localStorage", {
      value: {
        getItem: localStorageGetMock,
        setItem: localStorageSetMock,
      },
      writable: true,
    });

    const { container } = render(<App />);
    expect(localStorageGetMock).toHaveBeenCalledTimes(1);
  });
});
