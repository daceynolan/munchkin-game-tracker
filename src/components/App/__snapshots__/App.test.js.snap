// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders without crashing 1`] = `
<div
  class="game-content"
>
  <div
    class="header"
  >
    <img
      alt="Munchkin game logo"
      class="header__logo"
      src="logo-transparent.png"
    />
  </div>
  <div
    class="card"
  >
    <div
      class="card__inner"
      data-testid="card-inner"
    >
      <div
        class="splash-screen"
      >
        <img
          alt="Munchkin character holding chainsaw"
          class="splash-screen__character-image"
          src="main-character.png"
        />
        <a
          class="button button--default"
          data-testid="button"
          href="#/configure"
          type="button"
        >
          New Game
        </a>
      </div>
    </div>
  </div>
</div>
`;
